testutil/batch/manager.go

```diff
@@ -0,0 +1,612 @@
+	if err := m.PrecompileTxOnChainCheck(helperRecord.CurrentBatchID, msgType); err != nil {
+		logger.Error("EnqueueAndCheckTxsInBatch: failed to check whether the txs are on chain", "err", err)
+		return err
+	}
+	logger.Info("EnqueueAndCheckTxsInBatch: call txsCheckFunc", "msgType", msgType)
+	if err := txsCheckFunc(helperRecord.CurrentBatchID, msgType); err != nil {
+		logger.Error("EnqueueAndCheckTxsInBatch: failed to check the txs", "msgType", msgType, "err", err)
+		return err
+	}
+	// increase the batch id if the msg type is withdrawal
+	if msgType == assets.MethodWithdrawLST {
+		helperRecord.CurrentBatchID++
+		if err := SaveObject(m.GetDB(), helperRecord); err != nil {
+			logger.Error("EnqueueAndCheckTxsInBatch: can't save the helper record")
+			return err
+		}
+		logger.Info("EnqueueAndCheckTxsInBatch update the batchID", "batchID", helperRecord.CurrentBatchID)
+	}
+	return nil
+}
+
+func (m *Manager) WaitForShuttingDown() {
+	// Set up channel to listen for OS interrupt signals (like Ctrl+C)
+	stopChan := make(chan os.Signal, 1)
+	signal.Notify(stopChan, syscall.SIGINT, syscall.SIGTERM)
+
+	// Wait for an interrupt signal
+	<-stopChan
+	// Shutdown appManager gracefully
+	fmt.Println("Shutting down...")
+	close(m.Shutdown)
+	m.Close()
+}
+
+func (m *Manager) ExecuteBatchTestForType(msgType string) error {
+	if err := m.InitHelperRecord(); err != nil {
+		return err
+	}
+	// send test transactions in batch
+	go func() {
+		// deposit
+		if err := m.EnqueueAndCheckTxsInBatch(msgType); err != nil {
+			close(m.Shutdown)
+			return
+		}
+		logger.Info("ExecuteBatchTestForType: finish enqueuing and checking all txs")
+	}()
+	if err := m.DequeueAndSignSendTxs(); err != nil {
+		logger.Error("ExecuteBatchTestForType, failed to dequeue and sign send the txs,err:%s", err)
+		return err
+	}
+	return nil
+}
+
+func (m *Manager) Start() error {
+	if err := m.Prepare(); err != nil {
+		return err
+	}
+	if err := m.InitHelperRecord(); err != nil {
+		return err
+	}
+
+	// send test transactions in batch
+	go func() {
+		// deposit
+		if err := m.EnqueueAndCheckTxsInBatch(assets.MethodDepositLST); err != nil {
+			close(m.Shutdown)
+			return
+		}
+		logger.Info("Start: finish enqueuing and checking all deposit txs")
+		// delegation
+		if err := m.EnqueueAndCheckTxsInBatch(delegation.MethodDelegate); err != nil {
+			close(m.Shutdown)
+			return
+		}
+		logger.Info("Start: finish enqueuing and checking all delegation txs")
+		// undelegation
+		if err := m.EnqueueAndCheckTxsInBatch(delegation.MethodUndelegate); err != nil {
+			close(m.Shutdown)
+			return
+		}
+		logger.Info("Start: finish enqueuing and checking all undelegation txs")
+		// withdrawal
+		if err := m.EnqueueAndCheckTxsInBatch(assets.MethodWithdrawLST); err != nil {
+			close(m.Shutdown)
+			return
+		}
+		logger.Info("Start: finish enqueuing and checking all withdrawal txs")
+		time.Sleep(time.Duration(m.config.EachTestInterval) * time.Second)
+	}()
```

Comment:
Seems unnecessary, because it's only used for the test tool. And the next call `DequeueAndSignSendTxs` will block the progress. So how to use `eg.Wait`

Reply:
Thanks, I will change it in the next commit.